sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -351,11 +399,54 @@ export function getRawSqlParameters(parameters: SqlParameter[] | undefined): any
 /**
  * @internal
  * @ignore
- * Helper utility to build an instance of raw SQL parameter as `RawSqlParameter`
+ * Helper utility to extract array of userProperties key:value instances from given input,
+ * or undefined if not passed in.
+ * @param value
+ */
+export function getRawUserProperties(parameters: { [key: string]: any } | undefined): any {
+  if (parameters == undefined) {
+    return undefined;
+  }
+
+  const rawParameters: RawKeyValuePair[] = [];
+  for (let [key, value] of Object.entries(parameters)) {
+    let type: string | number;
+    if (typeof value === "number") {
+      type = TypeMapForRequestSerialization.number;
+    } else if (typeof value === "string") {
+      type = TypeMapForRequestSerialization.string;
+    } else {
+      throw new TypeError(
+        `Unsupported type for the value in the user property {${key}:${JSON.stringify(value)}}`
+      );
+      // TODO: Add more cases if other types are supported.
```

Comment:
You can remove this unless you mean you want to do this right now. I think it's obvious :)

Reply:
Doing followups

Reply:
File an issue for a future PR? Or is this something you want to do here?

Reply:
I'll log an issue with these details https://github.com/Azure/azure-sdk-for-js/pull/9794#discussion_r448111031

Reply:
I'll remove the comment here

Reply:
```suggestion
```